ReactNativeClient/lib/services/filterParser.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-cond-assign */
```

Comment:
I would prefer if you remove this, or perhaps only disable the line that's causing problem, explaining why it needs to be disabled.

Reply:
Disabling no-unused-vars was mostly for convenience and I saw it used in the existing searchEngine tests.
Disabling no-cond-assign was for the assignment statement in line 32.
`while ((match = filterRegex.exec(searchString))) `
I could rewrite it without this though.